sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,668 @@
+            component_diff[:, ids] += np.outer(exp_doc_topic_d, cnts / norm_phi)
+
+    return (doc_topic_distr, component_diff)
+
+
+class LatentDirichletAllocation(BaseEstimator, TransformerMixin):
+
+    """
+    Online Latent Dirichlet Allocation implementation with variational inference
+
+    Parameters
+    ----------
+    n_topics : int, optional (default=10)
+        Number of topics.
+
+    doc_topic_prior : float, optional (defalut=None)
+        Prior of document topic distribution `theta`. If the value is None, default
+        to `1 / n_topics`.
+        In the literature, this is called `alpha`.
+
+    topic_word_prior : float, optional (default=None)
+        Prior of topic word distribution `beta`. If the value is None, default
+        to `1 / n_topics`.
+        In the literature, this is called `eta`.
+
+    learning_method : 'batch' | 'online', default='online'
+        Method used to update `_component`. Only used in `fit` method.
+        In general, if the data size is large, online update will be much faster than
+        batch udpate.
+        Valid options::
+
+            'batch': Batch variational bayes method. Use all training data in each EM udpate.
+                Old `components_` will be overrided in each iteration.
+            'online': Online variational bayes method. In each EM update, use min batch of
+                training data to update `components_` variable incrementally. The weight in
+                each udpate is controlled by `learning_decay` and `learning_offset` parameter
+
+    learning_decay : float, optional (default=0.7)
+        It is a parameter that control learning weight for `_component` in online learning.
+        The value should be set between (0.5, 1.0] to guarantee asymptotic convergence.
+        When the value is 0.0 and batch_size is `n_samples`, the update is same as batch learning.
+        In the literature, this is called `kappa`.
+
+    learning_offset : float, optional (default=1000.)
+        A (positive) parameter that downweights early iterations in online learning.
+        It should be greater than 1.0. In the literature, this is called `tau0`.
+
+    max_iter : integer, optional (default=10)
+        The maximum number of iterations.
+
+    total_samples : int, optional (default=1e6)
+        Total umber of documents. It is only used in `partial_fit` method.
+
+    batch_size : int, optional (default=128)
+        Number of documents to use in each EM iteration. Only used in online learning method.
+
+    evaluate_every : int optional (default=-1)
+        How often to evaluate perplexity. Only used in `fit` method.
+        set it to -1 to not evalute perplexity in training at all.
+        Evaluating perplexity can help you check convergence in training process, but it
+        will also increase total training time.
+        Evaluating perplexity in every iteration will increase training time up to 2X.
+
+    normalize_doc : boolean, optional (default=False)
```

Comment:
Never mind, I thought this was used in fitting, but it is only a post-processing step.